digest/fd_digest.go

```diff
@@ -48,7 +49,7 @@ type fdWithDigest struct {
 
 func newFdWithDigest() FdWithDigest {
 	return &fdWithDigest{
-		digest: NewDigest(),
+		digest: adler32.New(),
```

Comment:
Since these aren't highly concurrent and we cache the use of them, we don't have to use the stack based adler32 approach - we can use the stdlib which is a few percent faster than the stack based adler32 approach (although stack based is zero-alloc, it'll only be fast in the highly concurrent use case).

Reply:
Cool, this probably warrants a comment somewhere in this package

Reply:
Yeah fair, I'll add some comments.